### DISCUSSION TITLE
log file and supply path tracing for large scale data

### DISCUSSION THREAD

**Discussion by maxware0715** [frePPLe/frepple#474] [Q&A] [answered]
I tried to implement production planning task by frepple, which is installed upon kubernetes and docker environment.

The data we use is about following amount:
Sales orders: <10k
Operation: 100k ~ 150k with lots of alter operation and item relation.

The engine seems to stuck in planning stage for several hours, while splitting sales orders into small batches costs less than 1 hour in sum. We try to tune "loglevel()" to 2 for issue tracing but it removed automatically after task is finished.

Could I get some tips to preserve the log file and get a guide to understand these logs?
Moreover, I need some advice about checking the supply path structure that may stuck the engine.

**Comment by jdetaeye** [comment, marked as answer]
>We try to tune "loglevel()" to 2 for issue tracing but it removed automatically after task is finished.

In the file /etc/frepple/djangosettings.py there is setting MAXTOTALLOGFILESIZE that defines a limit to the total disk space we allow for log files.  If this limit is reached, we automatically delete the oldest files.    With this feature we avoid keeping too many and too old log files.
Increasing the parameter will allow you to preserve the log file.

> Could I get some tips to preserve the log file and get a guide to understand these logs?
> Moreover, I need some advice about checking the supply path structure that may stuck the engine.

That's too open a question to answer... There are just too many possible causes, each with different alternative resolutions.

**Reply by maxware0715** [reply]
Thank you. For the second problem, I try to figure out which sales order may cost extremely long time. Which kind of log text should I focus?
